plasmapy/dispersion/analytical/stix_.py

```diff
@@ -0,0 +1,280 @@
+    >>> inputs = {
+    ...     "B": 8.3e-9 * u.T,
+    ...     "w": 0.001 * u.rad / u.s,
+    ...     "ions": [Particle("H+"), Particle("He+")],
+    ...     "n_i": [4.0e5,2.0e5] * u.m**-3,
+    ...     "theta": 30 * u.deg,
+    ... }
+    >>> stix(**inputs)
+    <Quantity [ 0.0...e+00-1.83219...e-07j,
+           -0.0...e+00+1.83219...e-07j,
+            1.83451...e-07-0.0...e+00j,
+           -1.83451...e-07+0.0...e+00j] rad / m>
+    """
+
+    # Validate ions argument
+    if not isinstance(ions, (list, tuple)):
+        ions = [ions]
+    ions = ParticleList(ions)
+
+    if not all(failed := [ion.is_ion and ion.charge_number > 0 for ion in ions]):
+        raise ValueError(
+            "Particle(s) passed to 'ions' must be a positively charged"
+            " ion. The following particle(s) is(are) not allowed "
+            f"{[ion for ion, fail in zip(ions, failed) if not fail]}"
+        )
+
+    # Validate n_i argument
+    if n_i.ndim not in (0, 1):
+        raise ValueError(
+            "Argument 'n_i' must be a single valued or a 1D array of "
+            f"size 1 or {len(ions)}, instead got shape of {n_i.shape}"
+        )
+    elif n_i.ndim == 1 and n_i.size != len(ions):
+        raise ValueError(
+            "Argument 'n_i' and 'ions' need to be the same length, got"
+            f" value of shape {len(ions)} and {len(n_i.shape)}."
+        )
+
+    n_i = n_i.value
+    if n_i.ndim == 0:
+        n_i = np.array([n_i] * len(ions))
+    elif n_i.size == 1:
+        n_i = np.repeat(n_i, len(ions))
+
+    species = ions + [Particle("e-")]
+    densities = np.zeros(n_i.size + 1)
+    densities[:-1] = n_i
+    densities[-1] = np.sum(n_i * ions.charge_number)
+
+    # Validate B argument
+    B = B.squeeze()
+    if B.ndim != 0:
+        raise ValueError(
+            "Argument 'B' must be single valued and not an array of"
+            f" shape  {B.shape}."
+        )
+
+    # Validate w argument and dimension
+    w = w.value.squeeze()
+    if w.ndim not in (0, 1):
+        raise ValueError(
+            "Argument 'w' needs to be a single value or a 1D array "
+            f" astropy Quantity, got a value of shape {w.shape}."
+        )
+    elif np.isscalar(w):
+        w = np.array([w])
+
+    # Validate theta value
+    theta = theta.value.squeeze()
+    if theta.ndim not in (0, 1):
+        raise TypeError(
+            "Argument 'theta' needs to be a single value or 1D array "
+            f" astropy Quantity, got array of shape {theta.shape}."
+        )
+    elif np.isscalar(theta):
+        theta = np.array([theta])
+
+    # Generate mesh grid of w x theta
+    w, theta = np.meshgrid(w, theta, indexing="ij")
+
+    # Generate the plasma parameters needed
+    wps = []
+    wcs = []
+    for par, dens in zip(species, densities):
+        wps.append(plasma_frequency(n=dens * u.m ** -3, particle=par).value)
+        wcs.append(gyrofrequency(B=B, particle=par, signed=False).value)
```

Comment:
```suggestion
        wcs.append(gyrofrequency(B=B, particle=par, signed=True).value)
```

Looking through Stix, the gyrofrequency he used is the the algebraic (signed) gyrofrequency.  Once I made this change, I was able to reproduce figure 1-1 in stix.

![image](https://user-images.githubusercontent.com/29869348/166491866-ff53031c-31c1-45f2-91fa-ac9a468e0cc2.png)